### DISCUSSION TITLE
Can we read STATIC_CRON_TOKEN from a file?

### DISCUSSION THREAD

**Discussion by shizunge** [firefly-iii/firefly-iii#5791] [General]
I have tried to read STATIC_CRON_TOKEN from a file specified by STATIC_CRON_TOKEN_FILE. It seems not working.

I know in .env.example, it does not mention that we can use STATIC_CRON_TOKEN_FILE, just try if can we use a file for any environment variable.

Can we add STATIC_CRON_TOKEN_FILE?

**Comment by JC5** [comment]
I [added this](https://dev.azure.com/Firefly-III/MainImage/_git/MainImage/commit/41fe0c05051c025f0e8cac691bfc1f7bb9261d60?refName=refs%2Fheads%2Fmain&path=%2Fentrypoint.sh&_a=compare) to the latest `:develop` tag, could you try it out?

**Comment by shizunge** [comment]
Thanks for the prompt reply.
It is working.

Does data-importer share the same infrastructure?

Could we also let data-importer read secrets (e.g. FIREFLY_III_ACCESS_TOKEN, NORDIGEN_KEY, SPECTRE_SECRET, AUTO_IMPORT_SECRET) from files?

**Reply by JC5** [reply]
It shouldn’t be hard, but it won’t be today. Let me put that on my list 👍
